src/main/java/org/springframework/shell/core/JLineShell.java

```diff
@@ -340,7 +342,7 @@ public void onShellStatusChange(final ShellStatus oldStatus, final ShellStatus n
 	}
 
 	private void flashMessageRenderer() {
-		if (!reader.getTerminal().isAnsiSupported()) {
+		if (!reader.getTerminal().isAnsiSupported() || disableColor) {
```

Comment:
can you pull this check out into a method called something like `usingColor`? these conditionals are getting pretty ugly